packages/components/src/progress-bar/index.tsx

```diff
@@ -46,6 +46,53 @@ function UnforwardedProgressBar(
 	);
 }
 
+/**
+ * A simple horizontal progress bar component.
+ *
+ * Supports two modes: determinate and indeterminate. A progress bar is determinate when a specific progress value has been specified (from 0 to 100), and indeterminate when a value hasn't been specified.
+ *
+ * ## Usage
+ *
+ * ```jsx
+ * import { ProgressBar } from '@wordpress/components';
+ *
+ * const MyLoadingComponent = () => {
+ *   return <ProgressBar />
+ * }
+ * ```
+ *
+ * The ProgressBar will expand to take all the available horizontal space of its immediate parent container element. To control its width, you can:
+ *
+ * Pass a custom CSS `className` that takes care of setting the `width`:
+ *
+ * ```css
+ * .my-css-class {
+ *   width: 160px;
+ * }
+ * ```
+ *
+ * ```jsx
+ * import { ProgressBar } from '@wordpress/components';
+ *
+ * const MyLoadingComponent = () => {
+ *   return <ProgressBar className="my-css-class" />;
+ * };
+ * ```
+ *
+ * Wrap it in a container element (e.g `<div>`) that has a `width` specified:
+ *
+ * ```jsx
+ * import { ProgressBar } from '@wordpress/components';
+ *
+ * const MyLoadingComponent = ( props ) => {
+ *   return (
+ *     <div style={ { width: '160px' } }>
+ *       <ProgressBar />
+ *     </div>
+ *   );
+ * };
+ * ```
```

Comment:
No hard rules, but in general we've been putting about this much in the component export JSDocs. A brief description, and a snippet that shows 1.) how to import it, and 2.) a basic usage example.

The main consumption points for this JSDoc data are IDE IntelliSense, and the Storybook Docs view (shown right below the default story).

```diff
 * A simple horizontal progress bar component.
 *
 * Supports two modes: determinate and indeterminate. A progress bar is determinate when a specific progress value has been specified (from 0 to 100), and indeterminate when a value hasn't been specified.
 *
-  * ## Usage
-  *
 * ```jsx
 * import { ProgressBar } from '@wordpress/components';
 *
 * const MyLoadingComponent = () => {
 *   return <ProgressBar />
 * }
 * ```
-  *
-  * The ProgressBar will expand to take all the available horizontal space of its immediate parent container element. To control its width, you can:
-  *
-  * Pass a custom CSS `className` that takes care of setting the `width`:
-  *
-  * ```css
-  * .my-css-class {
-  *   width: 160px;
-  * }
-  * ```
-  *
-  * ```jsx
-  * import { ProgressBar } from '@wordpress/components';
-  *
-  * const MyLoadingComponent = () => {
-  *   return <ProgressBar className="my-css-class" />;
-  * };
-  * ```
-  *
-  * Wrap it in a container element (e.g `<div>`) that has a `width` specified:
-  *
-  * ```jsx
-  * import { ProgressBar } from '@wordpress/components';
-  *
-  * const MyLoadingComponent = ( props ) => {
-  *   return (
-  *     <div style={ { width: '160px' } }>
-  *       <ProgressBar />
-  *     </div>
-  *   );
-  * };
-  * ```
```

Reply:
Ah cool, thanks for clarifying that, that makes sense 👍🏻